src/stores/ContractNamesStore.ts

```diff
@@ -16,7 +17,9 @@ const ContractNamesStore = proxy<ContractNamesStoreType>({
       return
     }
     const contract = ERC721__factory.connect(address, defaultProvider)
-    ContractNamesStore.contractNames[address] = contract.name()
+    ContractNamesStore.contractNames[address] = contract.callStatic
+      .name()
+      .catch(() => truncateMiddleIfNeeded(address, 17))
```

Comment:
Why 17? Also, this is a `ContractNamesStore`, not a `ContractNamesOrAddressesStore`. Return `undefined`, instead of the address if `name()` doesn't exist